pose_estimator/src/pose_estimator.cpp

```diff
@@ -32,361 +32,321 @@
   }
+  return 0;
+}
 
-  bool spin() {
-    while (!ros::isShuttingDown()) {
-      if (start() == 0) {
-        while (node_handle.ok()) {
-          ros::spin();
-        }
+bool PoseEstimatorNode::spin()
+{
+  while (!ros::isShuttingDown())
+  {
+    if (start() == 0)
+    {
+      while (node_handle.ok())
+      {
+        ros::spin();
       }
     }
-    stop();
-    return true;
-  }
-
-  void ahrsDataCallback(const sensor_msgs::Imu data) {
-    boost::mutex::scoped_lock lock(m_mutDataLock);
-    tf::Quaternion quat(data.orientation.x, data.orientation.y, data.orientation.z,
-                        data.orientation.w);
-    tf::Matrix3x3 m(quat);
-    double roll, pitch, yaw;
-    m.getRPY(roll, pitch, yaw);
-    cur_rpy_ang[CorrectedData::ROLL] = roll * -1.0;
-    cur_rpy_ang[CorrectedData::PITCH] = pitch * -1.0;
-    cur_rpy_ang[CorrectedData::YAW] = yaw;
-    // ROS_INFO("RPY: [%f,%f,%f]",roll*180.0/M_PI,pitch*180.0/M_PI,yaw*180.0/M_PI);
-    ahrs_ok = true;
   }
+  stop();
+  return true;
+}
 
-  void gpsDataCallback(const sensor_msgs::NavSatFix data) {
-    if (data.status.status >= 0) {
-      boost::mutex::scoped_lock lock(m_mutDataLock);
-      cur_latitude = data.latitude;
-      cur_longitude = data.longitude;
-      cur_gps_time = data.header.stamp.toSec();
-    }
-  }
+void PoseEstimatorNode::ahrsDataCallback(const sensor_msgs::Imu data)
+{
+  boost::mutex::scoped_lock lock(m_mutDataLock);
+  tf::Quaternion quat(data.orientation.x, data.orientation.y, data.orientation.z,
+                      data.orientation.w);
+  tf::Matrix3x3 m(quat);
+  double roll, pitch, yaw;
+  m.getRPY(roll, pitch, yaw);
+  cur_rpy_ang[CorrectedData::ROLL] = roll * -1.0;
+  cur_rpy_ang[CorrectedData::PITCH] = pitch * -1.0;
+  cur_rpy_ang[CorrectedData::YAW] = yaw;
+  ahrs_ok = true;
+}
 
-  void rpmDataCallback(const thruster_control::ReportRPM data) {
-    boost::mutex::scoped_lock lock(m_mutDataLock);
-    cur_speed = (double)data.rpms / rpmPerKnot;
-  }
+void PoseEstimatorNode::rpmDataCallback(const thruster_control::ReportRPM data)
+{
+  boost::mutex::scoped_lock lock(m_mutDataLock);
+  cur_speed = static_cast<double>(data.rpms) / rpmPerKnot;
+}
 
-  void pressureDataCallback(const sensor_msgs::FluidPressure data) {
+void PoseEstimatorNode::gpsDataCallback(const sensor_msgs::NavSatFix data)
+{
+  if (data.status.status >= 0)
+  {
     boost::mutex::scoped_lock lock(m_mutDataLock);
-    cur_pressure = data.fluid_pressure;
-    pressure_ok = true;
+    cur_latitude = data.latitude;
+    cur_longitude = data.longitude;
+    cur_gps_time = data.header.stamp.toSec();
   }
+}
 
-  void processDynPos() {
-    boost::mutex::scoped_lock data_lock(m_mutDataLock);
-    m_correctedData.rpy_ang.x = cur_rpy_ang[CorrectedData::ROLL];
-    m_correctedData.rpy_ang.y = cur_rpy_ang[CorrectedData::PITCH];
-    m_correctedData.rpy_ang.z = cur_rpy_ang[CorrectedData::YAW];
-  }
+void PoseEstimatorNode::pressureDataCallback(const sensor_msgs::FluidPressure data)
```

Comment:
@patapufate nit: make it a const ref 
```suggestion
void PoseEstimatorNode::pressureDataCallback(const sensor_msgs::FluidPressure& data)
```

Reply:
fixed